paddle/fluid/platform/nccl_helper.h

```diff
@@ -73,7 +74,9 @@ struct NCCLContextMap {
   std::unordered_map<int, NCCLContext> contexts_;
   std::vector<int> order_;
 
-  explicit NCCLContextMap(const std::vector<platform::Place> &places) {
+  explicit NCCLContextMap(const std::vector<platform::Place> &places,
+                          ncclUniqueId *nccl_id = nullptr,
+                          size_t node_count = 0, size_t trainer_id = 0) {
```

Comment:
node_count -> num_trainers?
should default num_trainers=1?

Reply:
> node_count -> num_trainers?

Done.

> should default num_trainers=1?

I think not, when `nccl_id != nullptr` we should be at nccl distributed mode which num_nodes always > 1

Reply:
But in single-machine set up, the actual number of trainers is 1? If a user see num_trainers=0 in the public interface of parallel_executor.py, he might just change that to 1 because he is actually using 1 node for training.